lib/segment/src/types.rs

```diff
@@ -4337,22 +4337,31 @@ fn shard_key_number_example() -> u64 {
 #[derive(Deserialize, Serialize, JsonSchema, Anonymize, Debug, Clone, PartialEq, Eq, Hash)]
 #[serde(untagged)]
 pub enum ShardKey {
-    #[schemars(example = "shard_key_string_example")]
-    Keyword(String),
+    #[schemars(
+        schema_with = "String::json_schema",
+        example = "shard_key_string_example"
+    )]
+    Keyword(SmolStr),
```

Comment:
Key change is here!

Both the stack allocation size of the enum type, and the OpenAPI schema remain the same.